### DISCUSSION TITLE
Navigator

### DISCUSSION THREAD

**Discussion by Tynopia** [Minestom/Minestom#1545] [Help]
![image](https://user-images.githubusercontent.com/65678882/204348823-f8f17cbf-a6d5-4ced-ac5e-261a92bceb6e.png)

My Fakeplayer often can't go over stairs or snow (not block).
The Fakeplayer stops and need few seconds (Often 5 - 10 sec) to find a new route.
Does anyone know a fix or workaround?

`fakePlayer.getNavigator().setPathTo(Storage.tutorialCoords.get(stage));`

**Comment by Timofffee** [comment]
The navigator uses a pathfinding algorithm that makes mistakes. This allows you to speed up the algorithm, but sometimes it creates the most pleasant things. Try other algorithm options via `Navigator.setPathFinder(..)`

**Comment by Tynopia** [comment]
Yes but i have no idea how i can fix it. I think i must change the shape in PFBlock... idk

**Comment by Obyvante** [comment]
I do believe current pathfinding is not that stable and some of the community members are building their own pathfinding system(like me). You may want to ask the community(discord or here) to see if there are any open sources or you may want to consider editing current pathfinding with your ideas and fixes.
